tests/data-type.js

```diff
@@ -141,6 +141,14 @@ test('test validity of `createdBy` field from another peer', async (t) => {
   await destroy()
 })
 
+test('test deletion of doc', async (t) => {
+  t.plan(1)
+  const projectKey = randomBytes(32)
+  const { dataType } = await testenv({ projectKey })
+  const { versionId } = await dataType.create(obsFixture)
+  t.execution(async () => await dataType.delete(versionId), `deletion failed`)
+})
```

Comment:
some suggestions for this:

1. personal preference but probably would rename the test to something like `delete()`

2. the assertion at the end has an incorrect message. the message should describe what's happening when the assertion succeeds i.e. it should say `deletion succeeded`. although in this case, the assertion is not strictly necessary (see 3)

3. you don't need to use `t.execution()` here. Instead would call it without being wrapped and use the returned value for additional checks (see 4)

4. We should have additional assertions about what the returned deleted doc looks like e.g. deleted and updateAt properties change

example implementation would look something like:

```ts
test('delete()', async (t) => {
  const projectKey = randomBytes(32)
  const { dataType } = await testenv({ projectKey })
  const createdDoc = await dataType.create(obsFixture)

  t.is(createdDoc.deleted, false, '`deleted` field is `false` before deletion')

  const deletedDoc = await dataType.delete(createdDoc.versionId)

  t.is(deletedDoc.deleted, true, '`deleted` field is `true` for after deletion')

  t.not(createdDoc.updatedAt, deletedDoc.updatedAt, 'updatedAt has changed')

  t.alike(
    deletedDoc.links,
    [createdDoc.versionId],
    'deleted doc links back to created doc'
  )

  const retrievedDocByDocId = await dataType.getByDocId(deletedDoc.docId)

  t.alike(
    retrievedDocByDocId,
    deletedDoc,
    'retrieving doc by doc id returns deleted doc'
  )
})
```

could potentially add an assertion for making sure that the `getByVersionId()` also returns the deleted doc, but ran into needing to do additional work because it returns a value with a `forks` field

Reply:
right, yeah, I'll add the comparison of the links and the retrieved `deleted` doc by docId